core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -142,13 +142,15 @@ class Partition(val topic: String,
 
   def delete() {
     // need to hold the lock to prevent appendMessagesToLeader() from hitting I/O exceptions due to log being deleted
+    val start = time.milliseconds
     inWriteLock(leaderIsrUpdateLock) {
       assignedReplicaMap.clear()
       inSyncReplicas = Set.empty[Replica]
       leaderReplicaIdOpt = None
       try {
-        logManager.deleteLog(TopicAndPartition(topic, partitionId))
+        logManager.asyncDelete(TopicAndPartition(topic, partitionId))
         removePartitionMetrics()
+        info("Deleted log for [%s,%d] in %d ms".format(topic, partitionId, (time.milliseconds - start)))
```

Comment:
- We should make it clear in this message that the file isn't actually deleted at this point. It is basically inaccessible/deleted from the user's POV and scheduled for removal.
- Here and elsewhere: rather than expand out the `[%s,%d]` format for the topic-partition we should just rely on the `toString` rendering of `TopicAndPartition`.
- Here and elsewhere: can also make this a little more concise with Scala's string interpolation.